google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/project/ProjectSelectionDialog.java

```diff
@@ -33,6 +103,346 @@
    */
   @Nullable
   CloudProject showDialog(@Nullable CloudProject cloudProject) {
-    return cloudProject;
+    // create dialog wrapper for the form once, create services and subscribe to login changes once.
+    if (dialogWrapper == null) {
+      dialogWrapper = new ProjectSelectionDialogWrapper();
+
+      dialogWrapper.setTitle(GctBundle.getString("cloud.project.selector.dialog.title"));
+      // disabled unless project is selected in the list.
+      dialogWrapper.setOKActionEnabled(false);
+      // install additional IDEA UI.
+      installTableSpeedSearch(projectListTable);
+
+      Stream.of(ProjectManager.getInstance().getOpenProjects())
+          .forEach(
+              project ->
+                  project
+                      .getMessageBus()
+                      .connect(dialogWrapper.getDisposable() /* disconnect once dialog is gone. */)
+                      .subscribe(
+                          GoogleLoginListener.GOOGLE_LOGIN_LISTENER_TOPIC,
+                          this::loadUsersAndProjects));
+    }
+
+    loadUsersAndProjects();
+    setCloudProject(cloudProject);
+
+    DialogManager.show(dialogWrapper);
+
+    int result = dialogWrapper.getExitCode();
+    return result == DialogWrapper.OK_EXIT_CODE ? getCloudProject() : null;
+  }
+
+  @VisibleForTesting
+  void setCloudProject(CloudProject cloudProject) {
+    if (cloudProject != null && !Strings.isNullOrEmpty(cloudProject.googleUsername())) {
+      Optional<CredentialedUser> loggedInUser =
+          Services.getLoginService().getLoggedInUser(cloudProject.googleUsername());
+      if (loggedInUser.isPresent()) {
+        selectedProjectsByAccount.put(loggedInUser.get(), cloudProject.projectName());
+        accountComboBox.setSelectedItem(loggedInUser.get());
+        updateProjectList(loggedInUser.get());
```

Comment:
why is this needed here if the project list is already loaded (and cached) in #loadUsersAndProjects ?

Reply:
oh..I think i see..thinking about this

Reply:
actually, nm. If the cached list is already loaded, why do we need to load it again for this specific user?

Reply:
It won't be loaded since it's cached, I sort of added it for increased readability so reader doesn't need to imply project will be selected because of the combox box event. We can remove it safely.

Reply:
ah ok. lets remove it then. it had the opposite effect on me. made me confused. :)

Reply:
done.